cli/popper/runner_slurm.py

```diff
@@ -0,0 +1,206 @@
+import os
+import tempfile
+import subprocess
+
+from popper import utils as pu
+from popper import scm
+from popper.cli import log as log
+from popper.runner import StepRunner as StepRunner
+from popper.runner_host import DockerRunner as HostDockerRunner
+from popper.runner_host import HostRunner
+
+
+class SlurmRunner(StepRunner):
+
+    spawned_jobs = set()
+
+    def __init__(self, config):
+        super(SlurmRunner, self).__init__(config)
+
+    def __exit__(self, exc_type, exc, traceback):
+        SlurmRunner.spawned_jobs = set()
+
+    def generate_script(self, cmd, job_id, job_script):
+        with open(job_script, "w") as f:
+            f.write("#!/bin/bash\n")
+            f.write(cmd)
+
+    def submit_batch_job(self, cmd, step):
+        job_id = pu.sanitized_name(step['name'], self.config.wid)
+        temp_dir = "/tmp/popper/slurm/"
+        os.makedirs(temp_dir, exist_ok=True)
+        job_script = os.path.join(temp_dir, f"{job_id}.sh")
+        out_file = os.path.join(temp_dir, f"{job_id}.out")
+        err_file = os.path.join(temp_dir, f"{job_id}.err")
+
+        self.generate_script(cmd, job_id, job_script)
+
+        sbatch_cmd = "sbatch --wait "
+        sbatch_cmd += f"--job-name {job_id} "
+        sbatch_cmd += f"--output {out_file} "
+        sbatch_cmd += f"--error {err_file} "
```

Comment:
since we're going to attach to the job, we can skip configuring output files and let that to be given by the user if they want